datajoint/schema.py

```diff
@@ -101,7 +102,12 @@ def spawn_missing_classes(self):
         in the context.
         """
         # if self.context is not set, use the calling namespace
-        context = self.context if self.context is not None else inspect.currentframe().f_back.f_locals
+        if self.context is not None:
+            context = self.context
+        else:
+            frame = inspect.currentframe().f_back
+            context = frame.f_locals
```

Comment:
Why not do the combination of locals and globals as was done in ERD init? In general, that combination is likely to yield the lookup of identifiers as expected by the user.

Reply:
This is inside `spawn_missing_classes` and the `context` variable is used for placing new objects, not just looking for existing objects.  So here it works a bit differently from other places.

Reply:
Ah, good point. I should have checked what the `context` was getting used for - we are actually modifying the local space. Thanks for the clarification.